internal/http/middleware/handler.go

```diff
@@ -1,145 +1,124 @@
-        log.Printf("Base64 error in segment starting at position %d: %v", i, err)
-        break
-      }
-    }
-
-    return c.Status(fiber.StatusUnauthorized).JSON(fiber.Map{
-      "message": "Invalid token encoding",
-      "error":   "The token contains invalid base64 data",
-      "detail":  err.Error(),
-    })
-  }
-
-  // Validate token
-  claims, err := secret.ValidateToken(tokenJwt)
-  if err != nil {
-    // Log the error to help with debugging
-    log.Printf("Token validation error (from %s): %v", tokenSource, err)
-
-    if errors.Is(err, secret.ErrTokenExpired) {
-      return c.Status(fiber.StatusUnauthorized).JSON(fiber.Map{
-        "message": "Token expired",
-      })
-    } else if errors.Is(err, secret.ErrEmptyToken) {
-      return c.Status(fiber.StatusUnauthorized).JSON(fiber.Map{
-        "message": "Empty token provided",
-      })
-    } else if errors.Is(err, secret.ErrInvalidToken) {
-      return c.Status(fiber.StatusUnauthorized).JSON(fiber.Map{
-        "message": "Invalid token",
-        "error":   err.Error(),
-      })
-    }
-
-    return c.Status(fiber.StatusUnauthorized).JSON(fiber.Map{
-      "message": "Authentication failed",
-      "error":   err.Error(),
-    })
-  }
-
-  // Store claims in context for use in route handlers
-  c.Locals("user_id", claims.UserID)
-  c.Locals("email", claims.Email)
-
-  return c.Next()
+	tokenJwt, err := checkToken(c)
+	if err != nil {
+		return c.Status(fiber.StatusUnauthorized).JSON(fiber.Map{
+			"message": "Unauthorized: No authentication token provided",
+			"error":   err.Error(),
+		})
+	}
+
+	// Validate token
+	claims, err := secret.ValidateToken(tokenJwt)
+	if err != nil {
+		if errors.Is(err, secret.ErrTokenExpired) {
+			return c.Status(fiber.StatusUnauthorized).JSON(fiber.Map{
+				"message": "Token expired",
+			})
+		} else if errors.Is(err, secret.ErrEmptyToken) {
+			return c.Status(fiber.StatusUnauthorized).JSON(fiber.Map{
+				"message": "Empty token provided",
+			})
+		} else if errors.Is(err, secret.ErrInvalidToken) {
+			return c.Status(fiber.StatusUnauthorized).JSON(fiber.Map{
+				"message": "Invalid token",
+				"error":   err.Error(),
+			})
+		}
+
+		return c.Status(fiber.StatusUnauthorized).JSON(fiber.Map{
+			"message": "Authentication failed",
+			"error":   err.Error(),
+		})
+	}
+
+	// Store claims in context for use in route handlers
+	c.Locals("user_id", claims.UserID)
+	c.Locals("email", claims.Email)
+
+	return c.Next()
+}
+
+func checkToken(c fiber.Ctx) (string, error) {
+	tokenJwt, err := checkCookieToken(c.Cookies("token"))
+	if err == nil && tokenJwt != "" {
+		return tokenJwt, nil
+	}
+
+	tokenJwt, err = checkAuthToken(c.Get("Authorization"))
+	if err == nil && tokenJwt != "" {
+		return tokenJwt, nil
+	}
+
+	return "", errors.New("no authentication token provided")
+}
+
+func checkCookieToken(tokenJwt string) (string, error) {
+	if tokenJwt != "" {
+		originalLength := len(tokenJwt)
+
+		if originalLength >= 4000 {
```

Comment:
For example this check. It's kinda useless at this point. If you really want to warn something like this you should warn it when you create the token. Not when you're validating.